packages/api/src/auth.js

```diff
@@ -145,14 +146,34 @@ export function withPinningAuthorized (handler) {
  */
 async function tryMagicToken (token, env) {
   let issuer = null
+  let tokenWasValidated = false
+  const isMagicTestMode = magicTestModeFromEnv(env)
+  const requiresTokenValidation = !isMagicTestMode
   try {
     env.magic.token.validate(token)
+    tokenWasValidated = true
+  } catch (error) {
+    switch (error.code) {
+      case 'ERROR_INCORRECT_SIGNER_ADDR':
+      case 'ERROR_MALFORMED_TOKEN':
+        break
+      default:
+        console.warn('unexpected error validating magic token: ', error.name, error.message)
+    }
+  }
+  const isAllowedTest = (env, token) => {
+    return Boolean(env.DANGEROUSLY_BYPASS_MAGIC_AUTH && token === 'test-magic')
```

Comment:
historical note: `DANGEROUSLY_BYPASS_MAGIC_AUTH` was [introduced as a workaround](https://github.com/web3-storage/web3.storage/pull/237#issuecomment-889772445) until magic-sdk landed support for test-mode. We can get rid of it now test-mode is a thing.

Reply:
I don't think we need both `NEXT_PUBLIC_MAGIC_TESTMODE_ENABLED` and `DANGEROUSLY_BYPASS_MAGIC_AUTH`. One way to bypass auth is one too many but a necessary horror.

Reply:
@olizilla I noticed that `DANGEROUSLY_BYPASS_MAGIC_AUTH` was there and mentioned that testmode wasn't out yet when it was authored.

However, I think actually these are worth being separate things... now that testMode exists and we can learn its limitations.

Specifically, the testMode stuff seems like it was released to be mostly about the *frontend* state managed by the magic sdk. e.g. it has some localstorage and stuff for its session. So when we're using it in the e2e tests, we need to construct `Magic` with `options.testMode === true` and that is what `NEXT_PUBLIC_MAGIC_TESTMODE_ENABLED` is for. Notably, that all works regardless of the value of `DANGEROUSLY_BYPASS_MAGIC_AUTH`. And when test-e2e uses those code paths, the magic tokens are syntactically correct magic tokens (iirc they are base64-encoded JSON strings).

But there is still some use for `DANGEROUSLY_BYPASS_MAGIC_AUTH`. When this is enabled, and only when this is enabled, the 'tokens' for magic auth are allowed to be 'test-magic'. Quite a few old unit tests use this token value and the tests only pass because `DANGEROUSLY_BYPASS_MAGIC_AUTH` is truthy.

So... I think both are useful and slightly different. There's not an easy way to remove `DANGEROUSLY_BYPASS_MAGIC_AUTH` without refactoring quite a few old tests. We can do that... but plz lets consider that as a separate issue.

Reply:
Since last comment, I did land this which provides a way for tests to not have to know the details of the 'test-magic' token (which only works when env.DANGEROUSLY_BYPASS_MAGIC_AUTH)
https://github.com/web3-storage/web3.storage/pull/1769

Reply:
Perhaps something more reasonable than a `DANGEROUSLY_BYPASS_MAGIC_AUTH` flag would be a new env var `WEB3_STORAGE_AUTHENTICATORS`

Most of the time
```
WEB3_STORAGE_AUTHENTICATORS=JSON.stringify({ oneOf: ['magic.link'] })
```

While testing:
```
WEB3_STORAGE_AUTHENTICATORS=JSON.stringify({
  oneOf: [
    'magic.link',
    {
      bearerTokens: ['test-magic'],
      capabilities: ['*'],
    },
  ]
})
```

At least this way it's just explicit extra configuration for 'oneOf the ways to authenticate a requestor', and not something 'dangerous' and explicitly having to do with disabling magic.link.

Reply:
The naming `DANGEROUSLY_BYPASS_MAGIC_AUTH` was deliberate! It should feel awkward and make us double check why and where we are setting it.

This proposal makes it look all sleek and like something that would be totally chill to set even in production!

Reply:
@olizilla In my last proposal defining `WEB3_STORAGE_AUTHENTICATORS`, there would no longer be a `DANGEROUSLY_BYPASS_MAGIC_AUTH`, so there would be no temptation to set (that env var) in production.